be/src/olap/comparison_predicate.h

```diff
@@ -180,6 +181,52 @@ class ComparisonPredicateBase : public ColumnPredicate {
                                bitmap);
     }
 
+    Status evaluate(const Schema& schema, InvertedIndexIterator* iterator,
+                            uint32_t num_rows, roaring::Roaring* bitmap) const override {
+        if (iterator == nullptr) {
+            return Status::OK();
+        }
+        auto column_desc = schema.column(_column_id);
+        std::string column_name = column_desc->name();
+
+        InvertedIndexQueryType query_type;
+        switch (PT)
+        {
+        case PredicateType::EQ:
+            query_type = InvertedIndexQueryType::EQUAL_QUERY;
+            break;
+        case PredicateType::NE:
+            query_type = InvertedIndexQueryType::EQUAL_QUERY;
+            break;
+        case PredicateType::LT:
+            query_type = InvertedIndexQueryType::LESS_THAN_QUERY;
+            break;
+        case PredicateType::LE:
+            query_type = InvertedIndexQueryType::LESS_EQUAL_QUERY;
+            break;
+        case PredicateType::GT:
+            query_type = InvertedIndexQueryType::GREATER_THAN_QUERY;
+            break;
+        case PredicateType::GE:
+            query_type = InvertedIndexQueryType::GREATER_EQUAL_QUERY;
+            break;
+        default:
+            return Status::InvalidArgument("invalid comparison predicate type {}", PT);
+        }
+
+        roaring::Roaring roaring;
+        RETURN_IF_ERROR(
+            iterator->read_from_inverted_index(column_name, &_value, query_type, num_rows, &roaring));
```

Comment:
If iterator is related to a column one by one, it should know the column_name and the arg column_name & schema is not needed.

Reply:
Besides, in MatchPredicate::evaluate, need the column type to distinguish between array and non array execute logic, input schema as parameter maybe better.